services/web/server/src/simcore_service_webserver/projects/_nodes_api.py

```diff
@@ -1,24 +1,71 @@
+import datetime
 import logging
 import mimetypes
 import urllib.parse
+from typing import Final
 
 from aiohttp import web
 from aiohttp.client import ClientError
+from models_library.projects import ProjectID
 from models_library.projects_nodes import Node, NodeID
 from models_library.projects_nodes_io import SimCoreFileLink
+from models_library.users import UserID
 from pydantic import (
     BaseModel,
     Field,
     HttpUrl,
+    NonNegativeFloat,
+    NonNegativeInt,
     ValidationError,
     parse_obj_as,
     root_validator,
 )
 
 from .._constants import APP_SETTINGS_KEY, RQT_USERID_KEY
+from ..application_settings import get_settings
 from ..storage.api import get_download_link
+from .exceptions import ProjectStartsTooManyDynamicNodes
 
 _logger = logging.getLogger(__name__)
+_NODE_START_INTERVAL_S: Final[datetime.timedelta] = datetime.timedelta(seconds=15)
+
+
+def get_service_start_lock_key(user_id: UserID, project_uuid: ProjectID) -> str:
```

Comment:
hmm... very much looking to find that refactoring in my PR...

Reply:
@sanderegg I wait for your PR to get in